File: python/example/tls.py
import dns

request = dns.Request()
request.Address = "dns.server"
request.Port = 853
request.Protocol = dns.Protocol.TLS
request.Qname = "github.com"
request.Type = dns.DNS_TYPE_A
request.Class = dns.Class.CH
request.Ca = "rootCA.crt"
request.Certificate = "certificate.pem"

response = dns.Resolv(request)
print(response) 

File: example/tls.c
#include <dns/dns.h>
#include <stdio.h>

int main(void)
{
    struct request r = { "dns.server", 853, tls, "github.com", DNS_TYPE_A, DNS_CLASS_IN, "rootCA.crt", "certificate.pem" };
    struct response resp;
    int result = resolv(r, &resp);
    if (result != 0) {
        perror("error");
    } else {
        print_response(resp);
        free_response(resp);
    }

    return 0;
}